param/parameterized.py

```diff
@@ -231,6 +238,70 @@ def _depends(*args,**kw):
     return _depends
 
 
+@accept_arguments
+def output(func, *output, **kw):
+    """
+    output allows annotating a method on a Parameterized class to
+    declare that it returns an output of a specific type. The outputs
+    of a Parameterized class can be queried using the
+    Parameterized.param.outputs method. By default the output will
+    inherit the method name but a custom name can be declared by
+    expressing the Parameter type using a keyword argument.
+
+    The simplest declaration simply declares the method returns an
+    object of without any Parameter type guarantees, e.g.:
+
+      @output()
+
+    If a specific parameter type is specified this is a declaration
+    that the method will return a value of that type, e.g.:
+
+      @output(param.Number())
+
+    To override the default name of the output the type may be declared
+    as a keyword argument, e.g.:
+
+      @output(custom_name=param.Number())
+
+    output also accepts Python object types which will be upgraded to
+    a ClassSelector, e.g.:
+
+      @output(int)
+    """
```

Comment:
```suggestion
    The object returned by this method will be validated against the specified type, raising an exception for values not accepted by that type.
    """
```

Reply:
Ah, I see above that you don't plan to do this validation soon?

Reply:
No, @jlstevens convinced me that it isn't immediately necessary and adding a validate method to all Parameters is a fair bit of work.

Reply:
Ok.